Relationships/RelationshipResolver.cs

```diff
@@ -221,18 +221,36 @@ private void Resolve(CkanModule module, RelationshipResolverOptions options)
                 return;
             }
 
-            foreach (string dep_name in stanza.Select(dep => dep.name))
+            foreach (var descriptor in stanza)
             {
+                string dep_name = descriptor.name;
                 log.DebugFormat("Considering {0}", dep_name);
 
                 // If we already have this dependency covered, skip.
                 // If it's already installed, skip.
-                if (modlist.ContainsKey(dep_name) || registry.IsInstalled(dep_name))
+
+                if (modlist.ContainsKey(dep_name))
```

Comment:
Note to self: This is a significant change. If we were previously treating installed module as satisfying dependencies, and now we're not, then we want to do extra testing where dependencies are installed in a separate pass, to make sure they still count in both the cmdline and the GUI clients.

It probably does make sense for the relationship resolver not to use the already installed modules, as long as whatever calls it has an easy way to add them in. If the clients themselves need to add in the list of installed mods, we should give them some sort of convenience method in the core.